### PR TITLE
fix(core): do not preventDefault when activeElement is select control

### LINKED ISSUE #6160: Bug(core): Select cannot be opened with keyboard
Due to https://github.com/SAP/fundamental-ngx/pull/5854/ the select popover options cannot be opened with keyboard.

### PR DESCRIPTION
**Pull request by mikerodonnell89**
Fixes #6160 

Fixes a bug where keyboard would no longer open the select popover body

### CODE REVIEW

**Review by droshev — APPROVED** [review]

File: libs/core/src/lib/popover/popover.component.ts [hunk 1]
```diff
@@ -178,7 +179,8 @@ export class PopoverComponent extends BasePopoverClass implements AfterViewInit,
     onKeyDown(event: KeyboardEvent): void {
         if (
             this.popoverControl.elRef.nativeElement.children[0] === document.activeElement &&
-            document.activeElement.tagName !== 'INPUT' && document.activeElement.tagName !== 'TEXTAREA'
+            document.activeElement.tagName !== 'INPUT' && document.activeElement.tagName !== 'TEXTAREA' &&
+            !document.activeElement.classList.contains(SELECT_CLASS_NAMES.selectControl)
```
Comment (Betrozov) [inline libs/core/src/lib/popover/popover.component.ts:183]:
```suggestion
            const { activeElement } = document;
            activeElement.tagName !== 'INPUT' && activeElement.tagName !== 'TEXTAREA' &&
            !activeElement.classList.contains(SELECT_CLASS_NAMES.selectControl)
```

**Review by Betrozov — CHANGES_REQUESTED** [review]

**Comment by kavya-b** [comment]
something strange happens after interacting with the mobile example:
- go to https://deploy-preview-6161--fundamental-ngx.netlify.app/#/core/select#mobile and click on the select input field to open the mobile view
- click on any item
- close the mobile view
- try to click on the select field in any other example, say [max height one](https://deploy-preview-6161--fundamental-ngx.netlify.app/#/core/select#max-height), it always opens the mobile view

File: libs/core/src/lib/popover/popover.component.ts [hunk 2]
```diff
@@ -176,9 +180,12 @@ export class PopoverComponent extends BasePopoverClass implements AfterViewInit,
     /** @hidden */
     @HostListener('keydown', ['$event'])
     onKeyDown(event: KeyboardEvent): void {
+        const activeElement = document.activeElement;
         if (
-            this.popoverControl.elRef.nativeElement.children[0] === document.activeElement &&
-            document.activeElement.tagName !== 'INPUT' && document.activeElement.tagName !== 'TEXTAREA'
+            this.popoverControl.elRef.nativeElement.children[0] === activeElement &&
+            activeElement.tagName !== 'INPUT' &&
```
Comment (aberikashvili) [inline libs/core/src/lib/popover/popover.component.ts:186]: May be it's better to extract these `INPUT`, `TEXTAREA` to constants?

**Review by Betrozov — APPROVED** [review]

File: libs/core/src/lib/popover/popover.component.ts (hunk 2 above)
Comment (mikerodonnell89) [reply libs/core/src/lib/popover/popover.component.ts:186]: I think we can count on these tagNames never changing so there is no added benefit of moving these to constants

**Pull request merged**